pymongo/mongo_client.py

```diff
@@ -40,6 +40,7 @@
     Any,
     Dict,
     FrozenSet,
+    Generator,
```

Comment:
The docs say: 
> Deprecated since version 3.9: [collections.abc.Generator](https://docs.python.org/3/library/collections.abc.html#collections.abc.Generator) now supports []. See [PEP 585](https://www.python.org/dev/peps/pep-0585) and [Generic Alias Type](https://docs.python.org/3/library/stdtypes.html#types-genericalias).

https://docs.python.org/3/library/typing.html#typing.Generator 

Do we need to use collections.abc.Generator on >=3.9?

Reply:
Done